### ISSUE TITLE
What is a benefit of using larger batch size?

### ISSUE THREAD

**Issue by John1231983** [matterport/Mask_RCNN#236]
Hello, I found that your implementation uses Batch Norm layer. The training BN layer often takes some advantage that requires a larger batch size. I am running your code on TitanX (12GB) and I can run with a batch size of `8`, the image size of `256x256` by changing `IMAGES_PER_GPU = 8`. Do you think that the changing batch size to 8 will have better performance? Now, I saw that it takes more 10 times consuming than training with the batch size of 1. In my opinion, we will have no benefit because your code did not train the BN. Is it right if I just set `training=True` to train the BN?
```
class BatchNorm(KL.BatchNormalization):
    """Batch Normalization class. Subclasses the Keras BN class and
    hardcodes training=False so the BN layer doesn't update
    during training.

    Batch normalization has a negative effect on training if batches are small
    so we disable it here.
    """

    def call(self, inputs, training=None):
        return super(self.__class__, self).call(inputs, training=False)
```

**Comment by JonathanCMitchell** [comment]
Hello @John1231983 I also have questions about this. It seems like the batch norm layers (when trainable = false) actually do nothing so I am not sure what their purpose is.

**Comment by waleedka** [comment]
You benefit from BN if the statistics that the BN layer collects over batches is a good approximation of the statistics of the entire dataset. This happens if the batch size is large (e.g., 32 or more ). If you use a small batch size, then BN might actually hurt performance. 

If you want to enable BN training, change the last line to:
```
    return super(self.__class__, self).call(inputs, training=training)
```

Don't hard-code it to `True`, because that will case the BN layer to run in training mode even when you do inferencing.

**Comment by GustavZ** [comment]
@waleedka so if i am training on small batch size how should i modify the config?
Is it even enough to change the conifg or should we remove the BN layers?

**Comment by EvanAlbee** [comment]
@gustavz  https://stackoverflow.com/questions/50047653/set-training-false-of-tf-layers-batch-normalization-when-training-will-get-a
